controllers/ocsinitialization/clusterclaims.go

```diff
@@ -0,0 +1,104 @@
+package ocsinitialization
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/storagecluster"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
+	extensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1alpha1 "open-cluster-management.io/api/cluster/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	ClusterClaimCRDName = "clusterclaims.cluster.open-cluster-management.io"
+)
+
+var (
+	ClusterClaimGroup              = "odf"
+	OdfInfoNamespacedNameClaimName = fmt.Sprintf("odfinfo.%s.openshift.io", ClusterClaimGroup)
+)
+
+type ocsClusterClaim struct{}
+
+type ClusterClaimCreator struct {
+	Context context.Context
+	Logger  logr.Logger
+	Client  client.Client
+	OcsInit *ocsv1.OCSInitialization
+}
+
+func doesClusterClaimCrdExist(ctx context.Context, client client.Client) (bool, error) {
+	crd := extensionsv1.CustomResourceDefinition{}
+	err := client.Get(ctx, types.NamespacedName{Name: ClusterClaimCRDName}, &crd)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (obj *ocsClusterClaim) ensureCreated(r *OCSInitializationReconciler, instance *ocsv1.OCSInitialization) (reconcile.Result, error) {
+	ctx := context.TODO()
+	if crdExists, err := doesClusterClaimCrdExist(ctx, r.Client); !crdExists {
+		if err != nil {
+			r.Log.Error(err, "An error has occurred while fetching customresourcedefinition", "CustomResourceDefinition", ClusterClaimCRDName)
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{}, nil
+	}
+
+	creator := ClusterClaimCreator{
+		Logger:  r.Log,
+		Context: ctx,
+		Client:  r.Client,
+		OcsInit: instance,
+	}
+
+	operatorNamespace, err := creator.getOperatorNamespace()
+	if err != nil {
+		r.Log.Error(err, "failed to get operator's namespace. retrying again")
+		return reconcile.Result{}, err
+	}
```

Comment:
No need of this. Simply fetch the operator namespace and proceed. This pattern was used earlier when there were a lot of things to fetch and create.